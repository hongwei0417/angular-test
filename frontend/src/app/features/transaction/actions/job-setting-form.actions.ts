import { createAction, props } from '@ngrx/store';

export const addJobSetting = createAction(
  '[JobSetting] add a job setting option'
);

export const removeJobSetting = createAction(
  '[JobSetting] remove a job setting option',
  props<{ id: number }>()
);
