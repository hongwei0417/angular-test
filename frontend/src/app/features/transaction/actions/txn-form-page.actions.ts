import { createAction, props, union } from '@ngrx/store';

export const createTxn = createAction(
  '[TxnFormPage] Create Transaction',
  props<{ title: string; content: string; executeCount: number }>()
);

export const clearTxnCreatePageState = createAction(
  '[TxnFormPage] Clear TxnCreate Page State'
);
