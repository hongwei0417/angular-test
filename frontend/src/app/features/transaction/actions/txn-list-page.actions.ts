import { createAction, props } from '@ngrx/store';

export const loadTxnList = createAction(
  '[TxnListPage] Load txn list table data'
);

export const clearTxnListPageState = createAction(
  '[TxnListPage] Clear txn list page state'
);

export const createNewTxn = createAction(
  '[TxnListPage] Create specific transaction info'
);

export const viewTxnInfo = createAction(
  '[TxnListPage] View specific transaction info',
  props<{ id: string }>()
);

export const editTxnInfo = createAction(
  '[TxnListPage] Edit specific transaction info',
  props<{ id: string }>()
);

export const copyTxnInfo = createAction(
  '[TxnListPage] Copy specific transaction info',
  props<{ id: string }>()
);

export const executeTxn = createAction(
  '[TxnListPage] Run specific transaction',
  props<{ id: string }>()
);

export const reloadTxn = createAction(
  '[TxnListPage] Reload specific transaction',
  props<{ id: string }>()
);
