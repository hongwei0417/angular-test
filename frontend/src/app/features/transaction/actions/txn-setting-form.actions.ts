import { createAction, props } from '@ngrx/store';

export const addFrequencySetting = createAction(
  '[TxnSetting] Add a frequency setting'
  // props<{ name: string }>()
);
