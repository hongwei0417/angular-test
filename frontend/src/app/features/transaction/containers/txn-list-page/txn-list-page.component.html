<h3 class="text-center">Transaction List Page</h3>

<div class="d-flex justify-content-center">
  <button [routerLink]="['/transaction', 'list']">Transaction list</button>
  <button [routerLink]="['/transaction', 'create']">Create Transaction</button>
</div>

<div class="d-flex justify-content-center">
  <button id="generateDataBtn" (click)="loadTxnData()">Generate Data</button>
</div>

<app-txn-list-table
  [tableData]="[]"
  (removeEvent)="onRemoveTxn($event)"
></app-txn-list-table>
