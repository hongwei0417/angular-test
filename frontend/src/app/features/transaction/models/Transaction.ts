export interface Transaction {
  id: string;
  title: string;
  content: string;
  executeCount: number;
  createTime?: Date;
}

export function generateMockTxn(): Transaction {
  return {
    id: '666',
    title: 'Testing',
    content: 'It is test content',
    executeCount: 10,
    createTime: new Date()
  };
}
