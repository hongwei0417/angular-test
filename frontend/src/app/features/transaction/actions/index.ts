import * as TxnApiActions from './txn-api.actions';
import * as TxnFormPageActions from './txn-form-page.actions';
import * as TxnListPageActions from './txn-list-page.actions';
import * as TxnSettingFormActions from './txn-setting-form.actions';

export {
  TxnApiActions,
  TxnFormPageActions,
  TxnListPageActions,
  TxnSettingFormActions,
};
