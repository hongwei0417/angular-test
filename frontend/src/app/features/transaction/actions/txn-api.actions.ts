import { ITransaction } from '../models/Transaction';
import { createAction, props, union } from '@ngrx/store';

export const loadTxnApiSuccess = createAction(
  '[TxnApi] Load Transaction Success',
  props<{ txns: ITransaction[] }>()
);

export const loadTxnApiFailure = createAction(
  '[TxnApi] Load Transaction Failure',
  props<{ error: any }>()
);

export const deleteTxnApiSuccess = createAction(
  '[TxnApi] Delete Transaction Failure',
  props<{ id: string }>()
);

export const createTxnApiSuccess = createAction(
  '[TxnApi] Create Transaction Success',
  props<{ txn: ITransaction }>()
);

export const createTxnApiFail = createAction(
  '[TxnApi] Create Transaction Failure',
  props<{ error: any }>()
);

const all = union({
  createTxnApiSuccess,
  createTxnApiFail,
  loadTxnApiSuccess,
  loadTxnApiFailure,
  deleteTxnApiSuccess,
});
export type TxnApiActionUnion = typeof all;
