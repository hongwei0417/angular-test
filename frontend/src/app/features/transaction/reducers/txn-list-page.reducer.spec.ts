import { generateMockTxn, Transaction } from './../models/Transaction';
import { TxnApiActions, TxnListPageActions } from '../actions';
import * as fromTxnList from './txn-list-page.reducer';
import { fakeData } from '../../../core/services/fake-data.service';

describe('Txn List Page Reducer', () => {
  describe('unknown action', () => {
    it('should return the previous state', () => {
      const action = {} as any;

      const result = fromTxnList.reducer(fromTxnList.initialState, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('load txn list', () => {
    it('should start loading txn data', () => {
      const action = TxnListPageActions.loadTxnList();
      const result = fromTxnList.reducer(fromTxnList.initialState, action);
      const newState: fromTxnList.State = {
        ...fromTxnList.initialState,
        loading: true,
      };
      expect(result).toEqual(newState);
      expect(result).not.toBe(fromTxnList.initialState);
    });

    it('should get all txn data', () => {
      const newState: fromTxnList.State = {
        ids: ['1', '2', '3'],
        entities: {
          1: fakeData[0],
          2: fakeData[1],
          3: fakeData[2],
        },
        loaded: true,
        loading: false,
        latestId: null,
      };
      const action = TxnApiActions.loadTxnApiSuccess({
        txns: fakeData,
      });
      const result = fromTxnList.reducer(fromTxnList.initialState, action);
      expect(result).toMatchSnapshot();
      expect(result).toEqual(newState);
    });
  });

  describe('create txn', () => {
    it('should get new txn data', () => {
      const payload = generateMockTxn();
      const action = TxnApiActions.createTxnApiSuccess({ txn: payload });
      const result = fromTxnList.reducer(fromTxnList.initialState, action);
      const newState = {
        ...fromTxnList.initialState,
        ids: ['666'],
        entities: {
          666: payload,
        },
      };
      expect(result).toEqual(newState);
    });
  });

  describe('remove txn', () => {
    it('should remove txn from state', () => {
      const removedId = '666';
      const initialState: fromTxnList.State = {
        ...fromTxnList.initialState,
        ids: ['666'],
        entities: {
          666: generateMockTxn(),
        },
      };
      const action = TxnApiActions.deleteTxnApiSuccess({ id: removedId });
      const result = fromTxnList.reducer(initialState, action);
      expect(result).toEqual(fromTxnList.initialState);
    });
  });
});
