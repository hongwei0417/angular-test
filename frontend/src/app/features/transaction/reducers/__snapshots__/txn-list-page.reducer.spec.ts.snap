// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Txn List Page Reducer an unknown action should return the previous state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "latestId": null,
  "loaded": false,
  "loading": false,
}
`;

exports[`Txn List Page Reducer load txn list should get all txn data 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "content": "test case 1",
      "createTime": 2021-08-23T16:00:00.000Z,
      "executeCount": 10,
      "id": "1",
      "title": "Kevin",
    },
    "2": Object {
      "content": "do something tonight...",
      "createTime": 2021-08-24T16:00:00.000Z,
      "executeCount": 4,
      "id": "2",
      "title": "Jack",
    },
    "3": Object {
      "content": "send email to everybody",
      "createTime": 2021-08-25T16:00:00.000Z,
      "executeCount": 1,
      "id": "3",
      "title": "Eric",
    },
  },
  "ids": Array [
    "1",
    "2",
    "3",
  ],
  "latestId": null,
  "loaded": true,
  "loading": false,
}
`;

exports[`Txn List Page Reducer load txn list should get all txn list data 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "latestId": null,
  "loaded": false,
  "loading": true,
}
`;

exports[`Txn List Page Reducer load txn list should start loading txn data 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "latestId": null,
  "loaded": false,
  "loading": true,
}
`;

exports[`Txn List Page Reducer unknown action should return the previous state 1`] = `
Object {
  "entities": Object {},
  "ids": Array [],
  "latestId": null,
  "loaded": false,
  "loading": false,
}
`;
